devtools/keypair/src/main.rs

```diff
@@ -1,47 +1,68 @@
-use rand::{rngs::OsRng, thread_rng, RngCore};
+#[macro_use]
+extern crate clap;
+
+use std::convert::TryFrom;
+use std::default::Default;
+
+use clap::App;
+use rand::{rngs::OsRng, RngCore};
+use serde::Serialize;
 use tentacle_secio::SecioKeyPair;
 
-use common_crypto::{BlsPrivateKey, PrivateKey, PublicKey, ToBlsPublicKey};
+use common_crypto::{BlsPrivateKey, PublicKey, ToBlsPublicKey};
 use protocol::types::{Address, Hash};
 use protocol::BytesMut;
 
+#[derive(Default, Serialize, Debug)]
+struct Keypair {
+    pub index:          u32,
+    pub private_key:    String,
+    pub public_key:     String,
+    pub address:        String,
+    pub bls_public_key: String,
+}
+
+#[derive(Default, Serialize, Debug)]
+struct Output {
+    pub common_ref: String,
+    pub keypairs:   Vec<Keypair>,
+}
+
 pub fn main() {
-    let mut seed = [0u8; 32];
-    OsRng.fill_bytes(&mut seed);
-
-    let seckey = Hash::digest(BytesMut::from(seed.as_ref()).freeze());
-    let keypair = SecioKeyPair::secp256k1_raw_key(seckey.as_bytes()).expect("secp256k1 keypair");
-    let pubkey = keypair.to_public_key().inner();
-    let user_addr = Address::from_pubkey_bytes(pubkey.into()).expect("user addr");
-
-    println!("seckey hex: {:?}", seckey.as_hex());
-    println!(
-        "pubkey hex: {:?}",
-        hex::encode(keypair.to_public_key().inner())
-    );
-    println!("user addr hex: {}", user_addr.as_hex());
-    println!("================================================================");
-
-    let n: usize = ::std::env::args()
-        .last()
-        .unwrap()
-        .parse()
-        .expect("argument error");
-    let common_ref = "muta";
-    println!("common ref: {:?}", hex::encode(common_ref.as_bytes()));
-
-    for i in 0..n {
-        let priv_key = BlsPrivateKey::generate(&mut thread_rng());
+    let yml = load_yaml!("keypair.yml");
+    let m = App::from(yml).get_matches();
+    let number = value_t!(m, "number", u32).unwrap();
+    let common_ref = m.value_of("common_ref").unwrap();
```

Comment:
I recommend not to write `common_ref` in the configuration file, but to randomly generate a string as `common_ref`. Then output the `common_ref` string to the console.